### PR TITLE
FK table search on data model admin screen

### PR DESCRIPTION
**Pull request by howonlee**
Pursuant to #13025.

Annoying thing that prevents you from just setting a param and being done with it is that the searches couldn't reach in deep within items in accordionlist search. So now they can also do that, and _then_ I just set a param and be done with it

### CODE REVIEW

File: frontend/src/metabase/components/AccordionList.jsx [hunk 1]
```diff
@@ -251,7 +262,9 @@ export default class AccordionList extends Component {
     let searchFilter = () => true;
     if (searchText) {
       searchFilter = item => {
-        let itemText = String(item[searchProp] || "");
+        let itemText = searchProp.includes(".")
+          ? this.getInItem(item, searchProp)
+          : String(item[searchProp] || "");
```
Comment (kulyk) [inline frontend/src/metabase/components/AccordionList.jsx:267]: We can use [`getIn` from `icepick`](https://github.com/aearly/icepick#getincollection-path) here:

```js
const path = searchProp.split(".")
let itemText = String(getIn(item, path) || "")
```

Would work for nested and not nested paths

**Review by kulyk — CHANGES_REQUESTED** [review]
Works great 👏 I'd only ask to use a library function instead of a custom one

**Comment by howonlee** [comment]
equivalent lib fn seems to exist only in lodash, not in underscore

**Comment by kulyk** [comment]
> equivalent lib fn seems to exist only in lodash, not in underscore

Take a look the code sample with [icepick](https://github.com/aearly/icepick#getincollection-path) I shared above

**Comment by howonlee** [comment]
ah, thats where the getIn can be found! cool beans will fix

**Review by flamber — COMMENTED** [review]
Perhaps I did a bad explanation in the issue, but this is what I expect:
1. Be able to search for the display_name of the **field** (currently it's searching for the database column name)
2. Be able to search for the display_name of the **table** (currently not possible)
3. Nice to have: If possible, also be able to search for the schema - that's not currently possible, but it's not possible to have much control over schema names.

Tested `b9f2852` and it now only allows searching for the table name (database name, not the display name). It's not possible to search for the field anymore.

Think of someone with 20 schemas, 100 tables in each schema, and 100 PKs (one in each table, but the underlying column is just called "id").

**Comment by howonlee** [comment]
that does make sense: i was thinking of the 'everything is named id in display name also' case

**Comment by howonlee** [comment]
This now does display_names over table and field and schema_name of table, but you can't search over multiple of these keys in the item at once: can't search `activity.id` by searching `activity.i` but `i` will work and `activ` will work

**Review by flamber — APPROVED** [review]
What a joy to actually be able to use the search!

File: frontend/src/metabase/components/AccordionList.jsx [hunk 2]
```diff
@@ -247,19 +268,29 @@ export default class AccordionList extends Component {
...
+          const searchResults = searchProp.map(member =>
+            this.searchPred(
+              item,
+              searchText,
+              member,
+              searchCaseInsensitive,
+              searchFuzzy,
+            ),
+          );
+          return searchResults.reduce((acc, curr) => acc || curr);
```
Comment (kulyk) [inline frontend/src/metabase/components/AccordionList.jsx:293]: `searchProp`, `searchCaseInsensitive` and `searchFuzzy` are props and `searchText` is state. Should we maybe extract them right in the `searchPred` method to keep this block more compact? I mean:

```js
searchPred = item => {
   const {searchProp, searchCaseInsensitive, searchFuzzy} = this.props;
   const {searchText} = this.state;
   // actual implementation
}
```

File: frontend/src/metabase/components/AccordionList.jsx [hunk 3]
```diff
@@ -247,19 +268,29 @@ export default class AccordionList extends Component {
...
-          return itemText.startsWith(searchText);
+        if (typeof searchProp === "string") {
+          return this.searchPred(
+            item,
+            searchText,
+            searchProp,
+            searchCaseInsensitive,
+            searchFuzzy,
+          );
+        } else if (searchProp.constructor === Array) {
```
Comment (kulyk) [inline frontend/src/metabase/components/AccordionList.jsx:283]: We should better use `Array.isArray(searchProp)` here

File: frontend/src/metabase/components/AccordionList.jsx [hunk 4]
```diff
@@ -223,6 +224,26 @@ export default class AccordionList extends Component {
     this.setState({ searchText });
   };
 
+  searchPred = (
```
Comment (kulyk) [inline frontend/src/metabase/components/AccordionList.jsx:227]: What does `pred` mean here? 🤔

**Review by kulyk — CHANGES_REQUESTED** [review]
List option for `searchProp` is _awesome_, thank you! Have just a few small comments here

File: frontend/src/metabase/components/AccordionList.jsx (hunk 4 above)
Comment (howonlee) [reply frontend/src/metabase/components/AccordionList.jsx:227]: predicate

File: frontend/src/metabase/components/AccordionList.jsx (hunk 2 above)
Comment (howonlee) [reply frontend/src/metabase/components/AccordionList.jsx]: need to pass in searchprop but otherwise sgtm

**Review by kulyk — APPROVED** [review]
LGTM, thank you for addressing the comments so diligently

**Pull request merged**